subsys/fs/nvs/nvs.c

```diff
@@ -298,7 +318,7 @@ static int nvs_ate_cmp_const(const struct nvs_ate *entry, uint8_t value)
 
 /* store an entry in flash */
 static int nvs_flash_wrt_entry(struct nvs_fs *fs, uint16_t id, const void *data,
-				size_t len)
+			       uint16_t len)
```

Comment:
Why uint16_t here?

Reply:
The offset is a 16 bit number. Directly taking it as uint16_t discloses the allowed range to the function caller and does not require any typecast in line 311/331.